www/updateUes.php

```diff
@@ -272,26 +311,37 @@ function updateUserExtSource($adapter, $userExtSource, $attributesToUpdate): boo
     return true;
 }
 
-function isSimpleType($attributeType): bool
+function isSimpleType($attribute): bool
 {
-    return strpos($attributeType, STRING_TYPE)
-    || strpos($attributeType, INTEGER_TYPE)
-    || strpos($attributeType, BOOLEAN_TYPE);
+    if (empty($attribute[TYPE])) {
+        return false;
+    }
+    $attributeType = $attribute[TYPE];
+
+    return strpos($attributeType, STRING_TYPE) ||
+        strpos($attributeType, INTEGER_TYPE) ||
+        strpos($attributeType, BOOLEAN_TYPE);
 }
 
-function isComplexType($attributeType): bool
+function isComplexType($attribute): bool
 {
+    if (empty($attribute[TYPE])) {
+        return false;
+    }
+    $attributeType = $attribute[TYPE];
+
     return strpos($attributeType, ARRAY_TYPE) ||
         strpos($attributeType, MAP_TYPE);
 }
 
-function convertToString($newValue)
+function convertToString(array $newValue): string
 {
+    $attrValueAsString = '';
+    Logger::debug('DFBDEBUG - SIMPLE TYPE before UNIQUE VALUES: ' . json_encode($newValue));
```

Comment:
DFBDEBUG :)
Please rewrite the log prefix or delete the row

Reply:
Ah good catch, removed the whole debug thing :)